
From: Jens Axboe <axboe@suse.de>

A stupid one-liner snuck into the modular io scheduler patch, please
apply this additional patch as well. Too much editing, sorry.

Signed-off-by: Jens Axboe <axboe@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/block/ll_rw_blk.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN drivers/block/ll_rw_blk.c~switchable-and-modular-io-schedulers-fix drivers/block/ll_rw_blk.c
--- 25/drivers/block/ll_rw_blk.c~switchable-and-modular-io-schedulers-fix	Thu Sep 30 15:44:35 2004
+++ 25-akpm/drivers/block/ll_rw_blk.c	Thu Sep 30 15:44:35 2004
@@ -2504,7 +2504,7 @@ void blk_wait_queue_drained(request_queu
 		}
 
 		finish_wait(&rl->drain, &wait);
-	} while (1);
+	}
 
 	spin_unlock_irq(q->queue_lock);
 }
_
